probe_scraper/probe_expiry_alert.py

```diff
@@ -44,93 +59,239 @@
+    return {"X-BUGZILLA-API-KEY": api_key}
+
+
+def get_bug_component(bug_id: int, api_key: str) -> Tuple[str, str]:
+    response = requests.get(BUGZILLA_BUG_URL + "/" + str(bug_id),
+                            headers=bugzilla_request_header(api_key))
+    response.raise_for_status()
+    bug = response.json()["bugs"][0]
+    return bug["product"], bug["component"]
+
+
+def find_existing_bugs(version: str, api_key: str) -> Set[str]:
+    search_query_params = {
+        "whiteboard": BUG_WHITEBOARD_TAG,
+        "include_fields": "description,summary",
+    }
+    response = requests.get(BUGZILLA_BUG_URL, params=search_query_params,
+                            headers=bugzilla_request_header(api_key))
+    response.raise_for_status()
+
+    found_bugs = response.json()["bugs"]
+
+    probes_with_bugs = set()
+    for bug in found_bugs:
+        if re.search(r"release: version (\d+)", bug["description"]).group(1) != version:
+            continue
+        probes_in_bug = re.search(r"```(.*)```", bug["description"], re.DOTALL).group(1).split()
+        for probe_name in probes_in_bug:
+            probes_with_bugs.add(probe_name)
+
+    return probes_with_bugs
+
+
+def get_longest_prefix(values: List[str], tolerance: int = 0) -> str:
+    """
+    Return the longest matching prefix among the list of strings.
+    If a prefix is less than 4 characters, return the first string.
+    Tolerance allows some characters to not match and returns the highest occurring prefix.
+    """
+    if tolerance < 0:
+        raise ValueError("tolerance must be >= 0")
+    if len(values) == 1:
+        return values[0]
+    if len(values) == 0:
+        return ""
+
+    if tolerance > 0:
+        longest_value_length = max(len(v) for v in values)
+        values = [v.ljust(longest_value_length) for v in values]
+
+    prefix_length = 0
+    for c in zip(*values):
+        if len(set(c)) > min([1 + tolerance, len(values) - 1]):
+            break
+        prefix_length += 1
+
+    if prefix_length < 4:
+        return values[0]
+
+    if tolerance == 0:
+        return values[0][:prefix_length]
+
+    prefix_count = defaultdict(int)
+    for value in values:
+        prefix_count[value[:prefix_length]] += 1
+
+    return sorted(prefix_count.items(), key=lambda item: item[1], reverse=True)[0][0] + '*'
+
+
+def create_bug(probes: List[ProbeDetails], version: str, api_key: str) -> int:
+    probe_names = [probe.name for probe in probes]
+    probe_prefix = get_longest_prefix(probe_names, tolerance=1)
+
+    create_params = {
+        "product": probes[0].product,
+        "component": probes[0].component,
+        "summary": BUG_SUMMARY_TEMPLATE.format(version=version, probe=probe_prefix),
+        "description": BUG_DESCRIPTION_TEMPLATE.format(
+            version=version, probes="\n".join(probe_names), notes=""),
+        "version": "unspecified",
+        "type": "task",
+        "whiteboard": BUG_WHITEBOARD_TAG,
+        "see_also": probes[0].previous_bug,
```

Comment:
In the examples all of them have the same See Also (the one for the first `browser.startup.*` bug that was filed). Is this an artefact of the dry run or is this evidence of a logic bug?

Reply:
That was just for testing where I used the same value because the actual bugs referenced don't exist in the dev bugzilla so it would cause an error.

Reply:
Oh good. That's my fears allayed.